tests/framework/extensions/machinepools/azure_machine_config.go

```diff
@@ -0,0 +1,73 @@
+package machinepools
+
+import (
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+const (
+	AzureKind                              = "AzureConfig"
+	AzurePoolType                          = "rke-machine-config.cattle.io.azureconfig"
+	AzureResourceConfig                    = "azureconfigs"
+	AzureMachingConfigConfigurationFileKey = "azureMachineConfig"
+)
+
+type AzureMachineConfig struct {
```

Comment:
Are we sure these are all the possible parameters the config could receive?
Nit: add comment to the specific upstream configuration

Reply:
Yes these are all the parameters. Is there something I am possibly missing Isaac? I just checked again on rancher, and this is it, unless I'm not seeing something. Other question what do you by "a comment to the specific upstream configuration" like a windows vs linux? Would be better in the actual test?

Reply:
Oh I thought this was coming from an upstream library but is it the Rancher API? In that case we're good.

Reply:
Yes this is in Rancher API, ok sounds good thanks for checking.